### DISCUSSION TITLE
Can I use `java.nio.MappedByteBuffer`  using StaticServlet?

### DISCUSSION THREAD

**Discussion by zhangtianxiao** [activej/activej#234] [Q&A]
`Bytebuf` can't seem to wrap `MappedByteBuffer`

**Comment by eduard-vasinskyi** [comment]
Hi, @zhangtianxiao

I’m not quite sure why you need to use MappedByteBuffer with StaticServlet. StaticServlet uses the [StaticLoader](https://github.com/activej/activej/blob/943d56f7f516f219b1846d4be346c22e6b49e6a8/core-http/src/main/java/io/activej/http/loader/StaticLoader.java) class which reads the file and already returns it as ByteBuf. 

If you want to save memory when loading the same file multiple times, you can use a cached StaticLoader created using the `StaticLoader.cacheOf()`,  `StaticLoader#cached()` methods.

You cannot _“wrap”_ a MappedByteBuffer into a ByteBuf, since ByteBuf uses a byte array under the hood. But if you want to “_convert”_ a MappedByteBuffer to a ByteBuf, you can do roughly the following:
```java
MappedByteBuffer buffer = ...;
byte[] bytes = new byte[buffer.remaining()];
buffer.get(bytes);
ByteBuf byteBuf = ByteBuf.wrapForReading(bytes);
```

**Reply by zhangtianxiao** [reply]
MappedByteBuffer will not copy memory to jvm heap, have better performance
